### ISSUE TITLE
Handle empty excel files

### ISSUE THREAD

**Issue by josecsotomorales** [elastacloud/spark-excel#42]
I had an issue with processing an empty file, should the lib return an empty dataframe instead?

```
nyc/tlc-trip/empty_excel.xlsx: java.lang.NullPointerException: Cannot invoke "elastashade.poi.ss.usermodel.Row.getRowNum()" because "firstDataRow.elem" is null
```

**Comment by josecsotomorales** [comment]
@dazfuller thoughts on this?

**Comment by dazfuller** [comment]
So the parser is designed to throw an exception saying that there is no data on the first row, which is what I'm seeing if I attempt to read a workbook with no data in it.

Is the excel file completely empty (I created a new Excel workbook, saved it, and closed it to test), and if so then can you share how you're reading it

**Comment by josecsotomorales** [comment]
1. 
[empty_excel.xlsx](https://github.com/user-attachments/files/16033199/empty_excel.xlsx)

**Comment by josecsotomorales** [comment]
I think it should return an empty dataframe with the field names on it.

**Comment by dazfuller** [comment]
Okay, I've been able to re-create it. It won't be able to return a completely empty data frame, but it can return a data frame with a single null record.

If the file is read by itself, then it would default to all string types for all fields (unless you provide a schema, because otherwise there's no data to infer one from). But if you read as a collection of files and it's further down the list, then it will be a null row which conforms to the inferred schema

**Comment by josecsotomorales** [comment]
That sounds good, thanks! 🚀

**Comment by dazfuller** [comment]
Okay, there's a version of that in the 0.1.13 release branch now

**Comment by dazfuller** [comment]
Fixed in the 0.1.13 release
